app/controllers/questionnaire_controller.rb

```diff
@@ -84,18 +84,39 @@ def check_interactive_content
     render(status: :bad_request, json: { error: 'Please supply a json string in the content field.' })
   end
 
+  # rubocop:disable Metrics/AbcSize
```

Comment:
damn, this function actually looks quite daunting. you're sure it does not make sense to split it up?

Reply:
I'll fix it.

Reply:
Fixed.